src/app/locator/qgslocatoroptionswidget.cpp

```diff
@@ -85,6 +90,35 @@ QgsLocatorFiltersModel::QgsLocatorFiltersModel( QgsLocator *locator, QObject *pa
 {
 }
 
+QWidget *QgsLocatorFiltersModel::configButton( const QModelIndex &index, QWidget *parent ) const
+{
+  if ( !index.isValid() )
+    return nullptr;
+
+  QgsLocatorFilter *filter = filterForIndex( index );
+  if ( filter && filter->hasConfigWidget() )
+  {
+    // use a layout to get the button center aligned
+    QWidget *w = new QWidget( parent );
+    QToolButton *bt = new QToolButton( );
+    QHBoxLayout *layout = new QHBoxLayout();
+    layout->setContentsMargins( 0, 0, 0, 0 );
+    layout->addWidget( bt );
+    w->setLayout( layout );
+
+    connect( bt, &QToolButton::clicked, this, [ = ]() {filter->openConfigWidget( bt );} );
+    //bt->setMinimumSize( 24, 24 );
+    bt->setMaximumSize( 24, 24 );
```

Comment:
This isn't going to work for hidpi displays - better to set using font metrics which resolve to just under 24 on low dpi and use max(fmSize, 24) to round to 24 on low dpi.